### ISSUE TITLE
KokkosKernels Test Failures on Hopper

### ISSUE THREAD

**Issue by csiefer2** [kokkos/kokkos-kernels#1891]
Build on lux27, using GCC 8.5.0, CUDA 11.8.

Nothing fancy in the cmake
```
cmake \
    -DKokkos_DIR=`pwd`/../build-kokkos/install/lib64/cmake/Kokkos \
    -DCMAKE_CXX_STANDARD=17 \
    -DCMAKE_BUILD_TYPE=RELEASE \
    -DCMAKE_INSTALL_PREFIX=`pwd`/install \
    -DKokkosKernels_ENABLE_TESTS=ON \
    -DKokkosKernels_ENABLE_EXAMPLES=ON \
    ../kokkos-kernels
```

Test failures:
```
The following tests FAILED:
	 11 - graph_cuda (Subprocess aborted)
	 13 - sparse_cuda (Subprocess aborted)
```

ctest --rerun-failed --output-on-failure output:
```
   Start 11: graph_cuda
1/2 Test #11: graph_cuda .......................Subprocess aborted***Exception:   6.32 sec
[==========] Running 12 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 12 tests from Cuda
[ RUN      ] Cuda.graph_graph_color_deterministic_default_scalar_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_deterministic_default_scalar_int_size_t_TestExecSpace (2 ms)
[ RUN      ] Cuda.graph_graph_color_distance2_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_distance2_double_int_size_t_TestExecSpace (147 ms)
[ RUN      ] Cuda.graph_graph_color_bipartite_sym_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_bipartite_sym_double_int_size_t_TestExecSpace (77 ms)
[ RUN      ] Cuda.graph_graph_color_bipartite_row_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_bipartite_row_double_int_size_t_TestExecSpace (93 ms)
[ RUN      ] Cuda.graph_graph_color_bipartite_col_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_bipartite_col_double_int_size_t_TestExecSpace (88 ms)
[ RUN      ] Cuda.graph_graph_color_default_scalar_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_default_scalar_int_size_t_TestExecSpace (198 ms)
[ RUN      ] Cuda.graph_graph_mis2_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_mis2_double_int_size_t_TestExecSpace (188 ms)
[ RUN      ] Cuda.graph_graph_mis2_coarsening_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_mis2_coarsening_double_int_size_t_TestExecSpace (102 ms)
[ RUN      ] Cuda.graph_random_graph_coarsen_double_int_size_t_TestExecSpace
cudaStreamSynchronize(stream) error( cudaErrorIllegalAddress): an illegal memory access was encountered /home/csiefer/kokkos-kernels/kokkos/core/src/Cuda/Kokkos_Cuda_Instance.cpp:151
Backtrace:
[0x61eee3] 
[0x617298] 
[0x6172fb] 
[0x626cbe] 
[0x62830d] 
[0x5bd267] 
[0x5bf7f9] 
[0x5bfc53] 
[0x440e1a] 
[0x506a37] 
[0x50ce93] 
[0x560c63] 
[0x426a3d] 
[0x5fc89a] 
[0x5ee807] 
[0x5eec75] 
[0x5eee26] 
[0x5f7cb6] 
[0x5f7ef5] 
[0x40e3f0] 
[0x7f9e4d9d7493] __libc_start_main
[0x41648e] 

   Start 11: graph_cuda
1/2 Test #11: graph_cuda .......................Subprocess aborted***Exception:   6.32 sec
[==========] Running 12 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 12 tests from Cuda
[ RUN      ] Cuda.graph_graph_color_deterministic_default_scalar_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_deterministic_default_scalar_int_size_t_TestExecSpace (2 ms)
[ RUN      ] Cuda.graph_graph_color_distance2_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_distance2_double_int_size_t_TestExecSpace (147 ms)
[ RUN      ] Cuda.graph_graph_color_bipartite_sym_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_bipartite_sym_double_int_size_t_TestExecSpace (77 ms)
[ RUN      ] Cuda.graph_graph_color_bipartite_row_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_bipartite_row_double_int_size_t_TestExecSpace (93 ms)
[ RUN      ] Cuda.graph_graph_color_bipartite_col_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_bipartite_col_double_int_size_t_TestExecSpace (88 ms)
[ RUN      ] Cuda.graph_graph_color_default_scalar_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_color_default_scalar_int_size_t_TestExecSpace (198 ms)
[ RUN      ] Cuda.graph_graph_mis2_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_mis2_double_int_size_t_TestExecSpace (188 ms)
[ RUN      ] Cuda.graph_graph_mis2_coarsening_double_int_size_t_TestExecSpace
[       OK ] Cuda.graph_graph_mis2_coarsening_double_int_size_t_TestExecSpace (102 ms)
[ RUN      ] Cuda.graph_random_graph_coarsen_double_int_size_t_TestExecSpace
cudaStreamSynchronize(stream) error( cudaErrorIllegalAddress): an illegal memory access was encountered /home/csiefer/kokkos-kernels/kokkos/core/src/Cuda/Kokkos_Cuda_Instance.cpp:151
Backtrace:
[0x61eee3] 
[0x617298] 
[0x6172fb] 
[0x626cbe] 
[0x62830d] 
[0x5bd267] 
[0x5bf7f9] 
[0x5bfc53] 
[0x440e1a] 
[0x506a37] 
[0x50ce93] 
[0x560c63] 
[0x426a3d] 
[0x5fc89a] 
[0x5ee807] 
[0x5eec75] 
[0x5eee26] 
[0x5f7cb6] 
[0x5f7ef5] 
[0x40e3f0] 
[0x7f9e4d9d7493] __libc_start_main
[0x41648e] 

```

**Comment by brian-kelley** [comment]
Luckily this a pretty straightforward fix - see #1927.

**Comment by bathmatt** [comment]
The KK bug fix referenced fixes this issue.
